<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Dalmolin Group" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Parameters - BulkRNA</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Parameters";
        var mkdocs_page_input_path = "params.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="..">
          <img src="../assets/BulkRNA_light.png" class="logo" alt="Logo"/>
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../usage/">Usage</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../output/">Output</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../CITATIONS/">Citations</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Reference</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" href="#">Parameters</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#inputoutput-options">Input/output options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#fastp-trimming-options">FastP trimming options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#kallisto-options">Kallisto options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#tximport-options">TXimport options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#institutional-config-options">Institutional config options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#max-job-request-options">Max job request options</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#generic-options">Generic options</a>
    </li>
    </ul>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">BulkRNA</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Reference</li>
      <li class="breadcrumb-item active">Parameters</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/dalmolingroup/bulkrna/edit/master/docs/params.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="dalmolingroupbulkrna-pipeline-parameters">dalmolingroup/bulkrna pipeline parameters</h1>
<p>Workflow for pre-processing, alignment and quantification of bulk RNA-Seq data</p>
<h2 id="inputoutput-options">Input/output options</h2>
<p>Define where the pipeline should find input data and save output data.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>input</code></td>
<td>Path to comma-separated file containing information about the samples in the experiment. <details><summary>Help</summary><small>You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 3 columns, and a header row.</small></details></td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>outdir</code></td>
<td>The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>email</code></td>
<td>Email address for completion summary. <details><summary>Help</summary><small>Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits. If set in your user config file (<code>~/.nextflow/config</code>) then you don't need to specify this on the command line for every run.</small></details></td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_title</code></td>
<td>MultiQC report title. Printed as page header, used for filename if not otherwise specified.</td>
<td><code>string</code></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="fastp-trimming-options">FastP trimming options</h2>
<p>Define fastp parameters for trimming</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>reads_minlength</code></td>
<td>reads shorter than length_required will be discarded</td>
<td><code>integer</code></td>
<td>15</td>
</tr>
<tr>
<td><code>fastp_adapter_fasta</code></td>
<td>specify a FASTA file to trim both read1 and read2 (if PE) by all the sequences in this FASTA file</td>
<td><code>string</code></td>
<td>https://gist.githubusercontent.com/jvfe/2d569adf47791b9d1e1b4ff810d410b8/raw/e970485d7fe2c9a5b501966eb26e9eb51789c86d/sample_adapters.fa</td>
</tr>
<tr>
<td><code>fastp_save_trimmed_fail</code></td>
<td>save reads that cannot pass the filters</td>
<td><code>boolean</code></td>
<td></td>
</tr>
<tr>
<td><code>fastp_save_merged</code></td>
<td>for paired-end input, merge each pair of reads into a single read if they are overlapped</td>
<td><code>boolean</code></td>
<td></td>
</tr>
<tr>
<td><code>fastp_qualified_quality</code></td>
<td>the quality value that a base is qualified</td>
<td><code>integer</code></td>
<td>15</td>
</tr>
<tr>
<td><code>fastp_cut_mean_quality</code></td>
<td>the mean quality requirement option shared by the cut_front and cut_tail sliding windows</td>
<td><code>integer</code></td>
<td>15</td>
</tr>
</tbody>
</table>
<h2 id="kallisto-options">Kallisto options</h2>
<p>Options for the alignment/quantification</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>transcriptome</code></td>
<td>Path to the transcriptome FASTA file to use for reference</td>
<td><code>string</code></td>
<td>None</td>
</tr>
<tr>
<td><code>index</code></td>
<td></td>
<td><code>string</code></td>
<td>None</td>
</tr>
<tr>
<td><code>fragment_length</code></td>
<td>Estimated average fragment length</td>
<td><code>integer</code></td>
<td>100</td>
</tr>
<tr>
<td><code>fragment_length_sd</code></td>
<td>Estimated standard deviation of fragment length</td>
<td><code>integer</code></td>
<td>1</td>
</tr>
</tbody>
</table>
<h2 id="tximport-options">TXimport options</h2>
<p>Options for importing Kallisto results into a count matrix</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>gtf</code></td>
<td></td>
<td><code>string</code></td>
<td>None</td>
</tr>
</tbody>
</table>
<h2 id="institutional-config-options">Institutional config options</h2>
<p>Parameters used to describe centralised config profiles. These should not be edited.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>custom_config_version</code></td>
<td>Git commit id for Institutional configs.</td>
<td><code>string</code></td>
<td>master</td>
</tr>
<tr>
<td><code>custom_config_base</code></td>
<td>Base directory for Institutional configs. <details><summary>Help</summary><small>If you're running offline, Nextflow will not be able to fetch the institutional config files from the internet. If you don't need them, then this is not a problem. If you do need them, you should download the files from the repo and tell Nextflow where to find them with this parameter.</small></details></td>
<td><code>string</code></td>
<td>https://raw.githubusercontent.com/nf-core/configs/master</td>
</tr>
<tr>
<td><code>config_profile_name</code></td>
<td>Institutional config name.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>config_profile_description</code></td>
<td>Institutional config description.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>config_profile_contact</code></td>
<td>Institutional config contact information.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>config_profile_url</code></td>
<td>Institutional config URL link.</td>
<td><code>string</code></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="max-job-request-options">Max job request options</h2>
<p>Set the top limit for requested resources for any single job.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>max_cpus</code></td>
<td>Maximum number of CPUs that can be requested for any single job. <details><summary>Help</summary><small>Use to set an upper-limit for the CPU requirement for each process. Should be an integer e.g. <code>--max_cpus 1</code></small></details></td>
<td><code>integer</code></td>
<td>16</td>
</tr>
<tr>
<td><code>max_memory</code></td>
<td>Maximum amount of memory that can be requested for any single job. <details><summary>Help</summary><small>Use to set an upper-limit for the memory requirement for each process. Should be a string in the format integer-unit e.g. <code>--max_memory '8.GB'</code></small></details></td>
<td><code>string</code></td>
<td>128.GB</td>
</tr>
<tr>
<td><code>max_time</code></td>
<td>Maximum amount of time that can be requested for any single job. <details><summary>Help</summary><small>Use to set an upper-limit for the time requirement for each process. Should be a string in the format integer-unit e.g. <code>--max_time '2.h'</code></small></details></td>
<td><code>string</code></td>
<td>240.h</td>
</tr>
</tbody>
</table>
<h2 id="generic-options">Generic options</h2>
<p>Less common options for the pipeline, typically set in a config file.</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
<th>Type</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>help</code></td>
<td>Display help text.</td>
<td><code>boolean</code></td>
<td></td>
</tr>
<tr>
<td><code>publish_dir_mode</code></td>
<td>Method used to save pipeline results to output directory. <details><summary>Help</summary><small>The Nextflow <code>publishDir</code> option specifies which intermediate files should be saved to the output directory. This option tells the pipeline what method should be used to move these files. See <a href="https://www.nextflow.io/docs/latest/process.html#publishdir">Nextflow docs</a> for details.</small></details></td>
<td><code>string</code></td>
<td>copy</td>
</tr>
<tr>
<td><code>email_on_fail</code></td>
<td>Email address for completion summary, only when pipeline fails. <details><summary>Help</summary><small>An email address to send a summary email to when the pipeline is completed - ONLY sent if the pipeline does not exit successfully.</small></details></td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>plaintext_email</code></td>
<td>Send plain-text email instead of HTML.</td>
<td><code>boolean</code></td>
<td></td>
</tr>
<tr>
<td><code>max_multiqc_email_size</code></td>
<td>File size limit when attaching MultiQC reports to summary emails.</td>
<td><code>string</code></td>
<td>25.MB</td>
</tr>
<tr>
<td><code>monochrome_logs</code></td>
<td>Do not use coloured log outputs.</td>
<td><code>boolean</code></td>
<td></td>
</tr>
<tr>
<td><code>hook_url</code></td>
<td>Incoming hook URL for messaging service <details><summary>Help</summary><small>Incoming hook URL for messaging service. Currently, only MS Teams is supported.</small></details></td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_config</code></td>
<td>Custom config file to supply to MultiQC.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_logo</code></td>
<td>Custom logo file to supply to MultiQC. File name must also be set in the MultiQC config file</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>multiqc_methods_description</code></td>
<td>Custom MultiQC yaml file containing HTML including a methods description.</td>
<td><code>string</code></td>
<td></td>
</tr>
<tr>
<td><code>tracedir</code></td>
<td>Directory to keep pipeline Nextflow logs and reports.</td>
<td><code>string</code></td>
<td>${params.outdir}/pipeline_info</td>
</tr>
<tr>
<td><code>validate_params</code></td>
<td>Boolean whether to validate parameters against the schema at runtime</td>
<td><code>boolean</code></td>
<td>True</td>
</tr>
<tr>
<td><code>show_hidden_params</code></td>
<td>Show all params when using <code>--help</code> <details><summary>Help</summary><small>By default, parameters set as <em>hidden</em> in the schema are not shown on the command line when a user runs with <code>--help</code>. Specifying this option will tell the pipeline to show all parameters.</small></details></td>
<td><code>boolean</code></td>
<td></td>
</tr>
</tbody>
</table>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../CITATIONS/" class="btn btn-neutral float-left" title="Citations"><span class="icon icon-circle-arrow-left"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/dalmolingroup/bulkrna" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../CITATIONS/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
