# dalmolingroup/bulkrna: Output

## Introduction

This document describes the output produced by the pipeline. Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

## Pipeline overview

The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:

- [FastQC](#fastqc) - Raw read QC
- [fastp](#fastp) - fastp reports (in HTML and JSON) about the trimming process
- [FastQC_trimmed](#fastqc) - Trimmed read QC - will be included in MultiQC results
- [Kallisto](#kallisto) - Kallisto quantification results
- [TxImport](#tximport) - Aggregate quantification results at gene and transcript levels
- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

### FastQC

<details markdown="1">
<summary>Output files</summary>

- `fastqc/`
  - `*_fastqc.html`: FastQC report containing quality metrics.
  - `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

[FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your sequenced reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. For further reading and documentation see the [FastQC help pages](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/).

![MultiQC - FastQC sequence counts plot](images/mqc_fastqc_counts.png)

![MultiQC - FastQC mean quality scores plot](images/mqc_fastqc_quality.png)

![MultiQC - FastQC adapter content plot](images/mqc_fastqc_adapter.png)

> **NB:** The FastQC plots displayed in the MultiQC report shows _trimmed_ reads (see [fastp](#fastp)).

### fastp

[fastp](https://github.com/OpenGene/fastp) is an ultra-fast all-in-one FASTQ preprocessor. Used in this workflow for adapter trimming and low-quality sequence
removal.

- `fastp`
  - `*.html` - HTML report generated by fastp
  - `*.json` - JSON report generated by fastp

### Kallisto

[`kallisto`](https://pachterlab.github.io/kallisto/manual) is used here
for aligning and quantifying targets against a reference transcriptome.

- `kallisto`
  - `{sample_id}` - Abundances quantified from said sample

### TxImport

[`tximport`](http://bioconductor.org/packages/release/bioc/html/tximport.html) is used here
to summarize transcript and gene-level counts from Kallisto.

- `tximport`
  - `gene_tpm.tsv` - Gene-level estimates (transcript-per-million) for the samples
  - `gene_counts.tsv` - Gene-level estimates (raw counts) for the samples
  - `transcript_tpm.tsv` - Transcript-level estimates (transcript-per-million) for the samples
  - `transcript_counts.tsv` - Transcript-level estimates (raw counts) for the samples
  - `transcript_scaled_tpm.tsv` - Transcript-level estimates (transcript-per-million) for the samples, scaled up to library size (scaledTPM)
  - `transcript_scaled_counts.tsv` - Transcript-level estimates (raw counts) for the samples, scaled up to library size (scaledTPM)

### MultiQC

<details markdown="1">
<summary>Output files</summary>

- `multiqc/`
  - `multiqc_report.html`: a standalone HTML file that can be viewed in your web browser.
  - `multiqc_data/`: directory containing parsed statistics from the different tools used in the pipeline.
  - `multiqc_plots/`: directory containing static images from the report in various formats.

</details>

[MultiQC](http://multiqc.info) is a visualization tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in the report data directory.

Results generated by MultiQC collate pipeline QC from supported tools e.g. FastQC. The pipeline has special steps which also allow the software versions to be reported in the MultiQC output for future traceability. For more information about how to use MultiQC reports, see <http://multiqc.info>.

### Pipeline information

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.
